kubernetes/upgrade/checks.go

```diff
@@ -172,10 +231,10 @@ func (checks *Checks) Run(ctx context.Context) error {
 }
 
 // PopulateRemovedCLIFlags populates the removed flags.
-func (e *ComponentRemovedItemsError) PopulateRemovedCLIFlags(node, component string, apiServerCLIFlags []string, removedFlags []string) {
+func (e *ComponentRemovedItemsError) PopulateRemovedCLIFlags(node, component string, cliFlags []string, removedFlags []string) {
 	for _, removedFlag := range removedFlags {
-		if slices.Contains(apiServerCLIFlags, func(s string) bool {
-			return strings.HasPrefix(s, "--"+removedFlag)
+		if slices.Contains(cliFlags, func(s string) bool {
+			return "--"+removedFlag == strings.Split(s, "=")[0]
```

Comment:
Fixes a bug in how cli flags were compared

Reply:
it might be eaiser to use `strings.Cut`, it's more to the point here I think